src/components/NavBar/NavBarModal.jsx

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import styles from './NavBarModal.module.scss';
+
+const NavBarModal = (props) => {
+    
+    return props.open ? ReactDOM.createPortal(
+        <div id="modal_navbar" className={styles.transparentModal} onClick={props.onClose}>
+        </div>,
+        document.body
+    ): null;
```

Comment:
You shouldn't be returning the result of `ReactDOM.createPortal` as it is undefined. This component can be simplified to an`if` statement.

Reply:
Unfortunately, the only way I got it to work is to embed it inside the render tree of NavBar